### PR TITLE
[DebugInfo] Set the autogenerated bit to request recycled locations

### PR DESCRIPTION
**Pull request by vedantk**
IRGenDebugInfo expects to see an autogenerated bit before allowing reuse
of the last seen debug location for a new instruction.

rdar://28544248

### CODE REVIEW

**Comment by vedantk** [comment]
@swift-ci Please smoke test

File: lib/IRGen/IRGenDebugInfo.cpp [hunk 1]
```diff
@@ -125,7 +125,7 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
   void finalize();
 
   void setCurrentLoc(IRBuilder &Builder, const SILDebugScope *DS,
-                     Optional<SILLocation> Loc = None);
+                     Optional<SILLocation> Loc);
```
Comment (adrian-prantl) [inline lib/IRGen/IRGenDebugInfo.cpp]: Is there a caller left that actually passes None? Otherwise we could drop the Optional.

**Review by adrian-prantl — APPROVED** [review]
Looks good!

File: lib/IRGen/IRGenDebugInfo.cpp (hunk 1 above)
Comment (vedantk) [reply lib/IRGen/IRGenDebugInfo.cpp]: Good call, I'll do that.

**Comment by vedantk** [comment]
@swift-ci Please smoke test and merge

**Comment by vedantk** [comment]
@swift-ci Please smoke test and merge

**Pull request merged**